### ISSUE TITLE
Expose full response in generateText and generateObject (Cannot access full groq errors)

### ISSUE THREAD

**Issue by williamlmao** [vercel/ai#3208]
### Feature Description

It would be really nice if generateText or generateObject could expose the full response, like so: 

```
 const { object, usage, fullResponse } = await generateObject({
    model,
    messages,
    schema,
  });
```

### Use Case

Certain inference providers like Groq put some additional information in other fields in an error response. For example if tool calls / json mode is on groq and the schema validation fails, it will tell you to look at the failed_generation. This is not currently accessible from the vercel AI SDK. 

### Additional context

If you want to see what I mean, just try bumping up the temperature to 2 on a generateObject from groq and use a somewhat complicated schema.

**Comment by williamlmao** [comment]
Closed in favor of #3779

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **None**  
Confidence: **High**

### Rationale
Although the underlying proposal is an API enhancement, this issue was explicitly closed as a duplicate in favor of another issue, so it is primarily duplicate bookkeeping.

### Evidence
- The request proposes exposing a new `fullResponse` field from `generateText` and `generateObject`.
- The sole comment states the issue was closed in favor of #3779.
